services/grafana/client.go

```diff
@@ -202,22 +214,60 @@ func (c *Client) getRole(ctx context.Context, authHeaders http.Header) (role, er
 		// check only default organization (with ID 1)
 		if id, _ := m["orgId"].(float64); id == 1 {
 			role, _ := m["role"].(string)
-			switch role {
-			case "Viewer":
-				return viewer, nil
-			case "Editor":
-				return editor, nil
-			case "Admin":
-				return admin, nil
-			default:
-				return none, nil
-			}
+			return c.convertRole(role), nil
 		}
 	}
 
 	return none, nil
 }
 
+func (c *Client) isApiKeyAuth(authHeader string) bool {
+	switch {
+	case strings.Contains(authHeader, "Bearer"):
+		return true
+	case strings.Contains(authHeader, "Basic"):
+		h := strings.TrimPrefix(authHeader, "Basic")
+		d, err := base64.StdEncoding.DecodeString(strings.TrimSpace(h))
+		if err != nil {
+			return false
+		}
+		return strings.HasPrefix(string(d), "api_key:")
+	}
+	return false
+}
+
+func (c *Client) convertRole(role string) role {
+	switch role {
+	case "Viewer":
+		return viewer
+	case "Editor":
+		return editor
+	case "Admin":
+		return admin
+	default:
+		return none
+	}
+}
+
+func (c *Client) getRoleforAPIKey(ctx context.Context, authHeaders http.Header) (role, error) {
+	var o map[string]interface{}
+	if err := c.do(ctx, "GET", "/api/org", "", authHeaders, nil, &o); err != nil {
```

Comment:
Now that we have a dedicated endpoint `/api/auth/key`, we don't need to make an extra call to `/api/org` (which felt like a code smell anyway).  So, I'd suggest we skip lines 253 trough 260.